### ISSUE TITLE
drush site-install fails with MySQL 8 (backport to drush8)

### ISSUE THREAD

**Issue by AdamPS** [drush-ops/drush#4259]
**Describe the bug**
See title.

**To Reproduce**
Run `drush site-install` with no database or user existing specifying a V8 mysql server.

**Expected behavior**
Database and user created

**Actual behavior**

> Failed to create database: ERROR 1064 (42000) at line 1: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IDENTIFIED BY '<PASSWORD>'' at line 1

**Workaround**
Create the user by hand first.

### System Configuration
Should be the same on any versions.  I was using drush 8, but I checked and the code is the same in master.

**Additional information**
The problem is that starting with MySQL 8 you no longer can (implicitly) create a user using the GRANT command.  . Use CREATE USER instead, followed by the GRANT statement.  See this [question/answer](https://stackoverflow.com/questions/50177216/how-to-grant-all-privileges-to-root-user-in-mysql-8-0).

**Comment by weitzman** [comment]
This has caused incompat with MariaDB in master branch so lets hold off on backport until thats resolved.

**Comment by mstenta** [comment]
> This has caused incompat with MariaDB in master branch so lets hold off on backport until thats resolved.

I just ran into this issue attempting to use Drush 8.3.5 `site-install` with MariaDB 10.5.5.

The query changes made in PR #4261 appear to be the cause, specifically the change to: `CREATE USER % IDENTIFIED WITH mysql_native_password BY '%s';`

I can confirm that that query alone does not work in MariaDB 10.5.5:

```
MariaDB [mysql]> CREATE USER test IDENTIFIED WITH mysql_native_password BY 'test';
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'BY 'test'' at line 1
```

**So as of Drush 8.3.0 (when #4261 was merged), `drush site-install` is broken on MariaDB.**

@totten's comment in https://github.com/drush-ops/drush/pull/4261#pullrequestreview-418924852 (which actually points to ANOTHER issue these queries may cause) includes a suggestion that may be a good option for fixing this: conditionally building the queries based on MySQL (5.6 vs 5.7) vs MariaDB.

**Comment by mstenta** [comment]
Oh just found https://github.com/drush-ops/drush/issues/4318 and https://github.com/drush-ops/drush/pull/4364 - looks like that may fix this issue?

Is that what you meant @weitzman ? Once #4364 is merged, we could backport the change to 8.x? If that's the next step here I'm happy to take a stab at it!

**Comment by mstenta** [comment]
Created a PR that backports #4364 to the 8.x branch. I simply did a `git cherry-pick cf1b639a2cabd55384180bdfbf022c04d3eb4c02`, and then resolved the conflicts:

- Modifies ` lib/Drush/Sql/Sqlmysql.php` instead of `src/Sql/SqlMysql.php`.
- Password variable is `$this->db_spec['password']` instead of `$dbSpec['password']`.

**Comment by mstenta** [comment]
PR #4548 has been merged. Does that mean we can close this?

**Comment by greg-1-anderson** [comment]
Yep

**Comment by mstenta** [comment]
@greg-1-anderson if there isn't anything else holding it up, do you think we could release 8.4.2 with this change? Thanks again!

**Comment by greg-1-anderson** [comment]
Done. (Releasing through GitHub / CI automation now)

**Comment by mstenta** [comment]
Thank you!!
